src/content/docs/start-here/kessel-inventory.md

```diff
@@ -0,0 +1,194 @@
+        string reporter_id
+        string reporter_type
+        string reporter_version
+        string subject_local_resource_id
+        string object_local_resource_id
+    }
+
+    Label["Label<JSON>"] {
+        string Key
+        string Value
+    }
+
+    Resource |o--o{ ResourceHistory : has
+    Relationship |o--o{ RelationshipHistory : has
+
+    Relationship }|--|| Resource : subject
+    Relationship }|--|| Resource : object
+
+    LocalToInventoryId ||--o{ Resource : maps
+
+%% Json maps
+Resource ||--|| ResourceReporter : reporter
+ResourceHistory ||--|| ResourceReporter : reporter
+
+Resource ||--o{ Label : labels
+ResourceHistory ||--o{ Label : labels
+
+Relationship ||--|| RelationshipReporter : reporter
+RelationshipHistory ||--|| RelationshipReporter : reporter
+```
+
+## Resources Lifecycle
+
+Inventory keeps the latest data for a resource, but it also keeps a history of the changes.
+
+When a resource is created, a new entry on the `resource_history` table is also created, indicated that the operation
+is `CREATE`.
+
+This value will match the operation that affected the result (`CREATE`, `UPDATE` or `DELETE`).
+Each entry in the `resource_history` may be deleted once it reach a certain time. 
+
+### Resource lifecycle diagram
+
+```mermaid
+flowchart LR
+%% Nodes
+    CreateResource("Create Resource")
+    CreateResourceExists{"Resource already reported?"}
+    CreateResourceDo("Adds resource")
+    CreateResourceHistory("Add resource_history(CREATE)")
+
+    UpdateResource("Update Resource")
+    UpdateResourceExists{"Resource already reported?"}
+    UpdateResourceDo("Updates resource")
+    UpdateResourceHistory("Add resource_history(UPDATE)")
+
+    DeleteResource("Delete Resource")
+    DeleteResourceExists{"Resource exists?"}
+    DeleteCopyResourceDo("Deletes resource")
+    DeleteResourceHistory("Add resource_history(DELETE)")
+    DeleteResourceRelationships("Deletes relations where this resource is the object or subject")
+
+    Fail("Fail")
+
+%% Links
+    
+    CreateResource --> CreateResourceExists
+    CreateResourceExists -- No --> CreateResourceDo --> CreateResourceHistory
+    CreateResourceExists -- Yes --> Fail
+
+    UpdateResource --> UpdateResourceExists
+    UpdateResourceExists -- No --> CreateResourceDo
+    UpdateResourceExists -- Yes --> UpdateResourceDo --> UpdateResourceHistory
+
+    DeleteResource --> DeleteResourceExists
+    DeleteResourceExists -- No --> Fail
+    DeleteResourceExists -- Yes --> DeleteCopyResourceDo --> DeleteResourceHistory --> DeleteResourceRelationships
+```
+
+### Relationship lifecycle
+
+The lifecycle of a relationship is similar to a resource, as there is a history table where changes on the relationship are tracked.
+It follows the same approach of indicating the operation (`CREATE`, `UPDATE` and `DELETE`). 
+Note than when a resource is deleted, the relationship is also deleted, but without updating the history table for that.
```

Comment:
The history table is updated at the DELETE event though, right?

Reply:
When we deleted a `resource`, the `resource_history` table gets a new `DELETED` entry. `relationships_history` table does not get any entry in this case.
When we directly delete a `relationship`, `relationship_history` tables do get a `DELETED` entry.

I'll update above to make it clear.